mne/annotations.py

```diff
@@ -647,6 +647,186 @@ def rename(self, mapping, verbose=None):
         return self
 
 
+class EpochAnnotationsMixin:
+    """Mixin class for Annotations in Epochs."""
+
+    @property
+    def annotations(self):  # noqa: D102
+        return self._annotations
+
+    def set_annotations(self, annotations):
+        """Setter for Epoch annotations.
+
+        Parameters
+        ----------
+        annotations : instance of mne.Annotations | None
+            Annotations to set.
+
+        Returns
+        -------
+        self : instance of Raw
+            The raw object with annotations.
+
+        Notes
+        -----
+        Epoch Annotations were added in v1.0 in order to preserve
+        events that occurred in raw data when they were Epoched. For example,
+        this might occur in time-frequency analysis, or sliding-window
+        connectivity analysis. Setting Annotations directly on Epochs
+        requires an accurate ``_raw_sfreq``, which stores the raw sampling
+        rate to interpret the Annotation.
+        """
+        self._set_annotations(annotations)
+        return self
```

Comment:
I believe the main reason I had a private method was to prevent users from calling `set_annotations`, but now that we have a public method and we sufficiently warn the user of adding annotations after resampling, then I think we can get rid of the private method.

Reply:
Removed private method.